game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/AggregateResults.java

```diff
@@ -13,30 +13,79 @@
 import java.util.Optional;
 import lombok.Getter;
 import lombok.Setter;
+import org.apache.commons.math3.stat.descriptive.moment.Mean;
+import org.apache.commons.math3.stat.descriptive.moment.StandardDeviation;
 import org.triplea.java.collections.IntegerMap;
 import org.triplea.util.Tuple;
 
-/** A container for the results of multiple battle simulation runs. */
+/**
+ * A container for the results of multiple battle simulation runs.
+ *
+ * <p>This class wraps a collection of {@code BattleResult}s and provides methods to query certain
+ * statistical properties, e.g. the win probability, or the standard deviation for the number of
+ * unit left.
+ *
+ * <p>This class does not restrict the battle result to come from the same battle setup. If this is
+ * desired, the user must ensure that this property hold.
+ *
+ * <p>Note on bias correction: Given a set of data points there are different variances (or standard
+ * deviations). If the data set for which the variance is calculated contains not all possible
+ * observations/states, i.e. is a subset, then bias correction should be applied. This is because
+ * the variance needs the expected value/mean, however, since we do not have all observations the
+ * calculated mean has an error which propagates to the variance. To mitigate this, a bias corrected
+ * variance is used. For more information see any book about statistics or [1].
+ *
+ * <p>[1] https://en.wikipedia.org/wiki/Variance#Population_variance_and_sample_variance
+ */
 public class AggregateResults {
   private final List<BattleResults> results;
   @Getter @Setter private long time;
 
+  /**
+   * Creates a new aggregator and sets the internal storage size to {@code expectedCount}. Choosing
+   * a good estimate reduces the number of reallocations of the internal storage once it hits its
+   * limit.
+   *
+   * @param expectedCount number of expected results to store
+   */
   public AggregateResults(final int expectedCount) {
     results = new ArrayList<>(expectedCount);
   }
 
-  AggregateResults(final List<BattleResults> results) {
+  /**
+   * Creates a new aggregator and populates it with thr battle results {@code results}.
+   *
+   * @param results the battle results to add to this aggregator.
+   */
+  public AggregateResults(final List<BattleResults> results) {
     this.results = new ArrayList<>(results);
   }
 
+  /**
+   * Add the battle result {@code result} to this aggregator.
+   *
+   * @param result the battle result to add.
+   */
   public void addResult(final BattleResults result) {
     results.add(result);
   }
 
+  /**
+   * Add all battle results in {@code results} to this aggregator.
+   *
+   * @param results the battle results to add
+   */
   public void addResults(final Collection<BattleResults> results) {
     this.results.addAll(results);
   }
 
+  /**
```

Comment:
nit, this block can be replaced with a lombok `@Getter` on the variable, then we don't need to worry about a javadoc or writing the 'get' method by hand either.

Reply:
Will do.  However, I noticed you squash all commits into one, so I'll postpone this into another pull request – probably together with the javadoc update – as they are not related to implementing the standard deviation.